src/Mod/Sketcher/Gui/DrawSketchHandlerScale.h

```diff
@@ -0,0 +1,529 @@
+ *   GNU Library General Public License for more details.                  *
+ *                                                                         *
+ *   You should have received a copy of the GNU Library General Public     *
+ *   License along with this library; see the file COPYING.LIB. If not,    *
+ *   write to the Free Software Foundation, Inc., 59 Temple Place,         *
+ *   Suite 330, Boston, MA  02111-1307, USA                                *
+ *                                                                         *
+ ***************************************************************************/
+
+
+#ifndef SKETCHERGUI_DrawSketchHandlerScale_H
+#define SKETCHERGUI_DrawSketchHandlerScale_H
+
+#include <QApplication>
+
+#include <Gui/BitmapFactory.h>
+#include <Gui/Notifications.h>
+#include <Gui/Command.h>
+#include <Gui/CommandT.h>
+
+#include <Mod/Sketcher/App/GeometryFacade.h>
+#include <Mod/Sketcher/App/SketchObject.h>
+
+#include "DrawSketchDefaultWidgetController.h"
+#include "DrawSketchControllableHandler.h"
+
+#include "GeometryCreationMode.h"
+#include "Utils.h"
+
+using namespace Sketcher;
+
+namespace SketcherGui
+{
+
+extern GeometryCreationMode geometryCreationMode;  // defined in CommandCreateGeo.cpp
+
+class DrawSketchHandlerScale;
+
+using DSHScaleController =
+    DrawSketchDefaultWidgetController<DrawSketchHandlerScale,
+                                      StateMachines::ThreeSeekEnd,
+                                      /*PAutoConstraintSize =*/0,
+                                      /*OnViewParametersT =*/OnViewParameters<3>,
+                                      /*WidgetParametersT =*/WidgetParameters<0>,
+                                      /*WidgetCheckboxesT =*/WidgetCheckboxes<1>,
+                                      /*WidgetComboboxesT =*/WidgetComboboxes<0>>;
+
+using DSHScaleControllerBase = DSHScaleController::ControllerBase;
+
+using DrawSketchHandlerScaleBase = DrawSketchControllableHandler<DSHScaleController>;
+
+class DrawSketchHandlerScale: public DrawSketchHandlerScaleBase
+{
+    friend DSHScaleController;
+    friend DSHScaleControllerBase;
+
+public:
+    explicit DrawSketchHandlerScale(std::vector<int> listOfGeoIds)
+        : listOfGeoIds(listOfGeoIds)
+        , deleteOriginal(true)
+        , refLength(0.0)
+        , length(0.0)
+        , scaleFactor(0.0)
+    {}
+
+    DrawSketchHandlerScale(const DrawSketchHandlerScale&) = delete;
+    DrawSketchHandlerScale(DrawSketchHandlerScale&&) = delete;
+    DrawSketchHandlerScale& operator=(const DrawSketchHandlerScale&) = delete;
+    DrawSketchHandlerScale& operator=(DrawSketchHandlerScale&&) = delete;
+
+    ~DrawSketchHandlerScale() override = default;
+
+private:
+    void updateDataAndDrawToPosition(Base::Vector2d onSketchPos) override
+    {
+        switch (state()) {
+            case SelectMode::SeekFirst: {
+                referencePoint = onSketchPos;
+                referencePoint = onSketchPos;

```

Comment:
Once was not enough?

Reply:
Ah I forgot about these, I will fix them.